pandaserver/dataservice/adder_gen.py

```diff
@@ -117,411 +117,461 @@ def get_plugin_class(self, tmp_vo, tmp_group):
+                            associate_job = self.taskBuffer.peekJobs(
+                                [associate_job_id],
+                                fromDefined=False,
+                                fromArchived=False,
+                                fromWaiting=False,
+                                forAnal=True,
+                            )[0]
+                            if self.job is None:
+                                self.logger.debug(f": associated job PandaID={associate_job_id} not found in DB")
                             else:
-                                closer_thread = closer.Closer(self.taskBuffer, destination_dispatch_block_list, self.job)
-                            self.logger.debug("start Closer")
-                            closer_thread.run()
-                            del closer_thread
-                            self.logger.debug("end Closer")
-                        # run closer for associate parallel jobs
-                        if EventServiceUtils.isJobCloningJob(self.job):
-                            associate_dispatch_block_map = self.taskBuffer.getDestDBlocksWithSingleConsumer(
-                                self.job.jediTaskID, self.job.PandaID, destination_dispatch_block_list
-                            )
-                            for associate_job_id in associate_dispatch_block_map:
-                                associate_dispatch_blocks = associate_dispatch_block_map[associate_job_id]
-                                associate_job = self.taskBuffer.peekJobs(
-                                    [associate_job_id],
-                                    fromDefined=False,
-                                    fromArchived=False,
-                                    fromWaiting=False,
-                                    forAnal=True,
-                                )[0]
-                                if self.job is None:
-                                    self.logger.debug(f": associated job PandaID={associate_job_id} not found in DB")
-                                else:
-                                    closer_thread = closer.Closer(self.taskBuffer, associate_dispatch_blocks, associate_job)
-                                    self.logger.debug(f"start Closer for PandaID={associate_job_id}")
-                                    closer_thread.run()
-                                    del closer_thread
-                                    self.logger.debug(f"end Closer for PandaID={associate_job_id}")
+                                closer_thread = closer.Closer(self.taskBuffer, associate_dispatch_blocks, associate_job)
+                                self.logger.debug(f"start Closer for PandaID={associate_job_id}")
+                                closer_thread.run()
+                                del closer_thread
+                                self.logger.debug(f"end Closer for PandaID={associate_job_id}")
+
+
+    # main
+    def run(self):
```

Comment:
I would like a bit more order, for example:
- run() should be at the beginning of the class or at the end. Now it's in the middle. 
- Other functions like get_plugin_class and execute_plugin could also be together, instead of spread separately around.
- Functions about getting or parsing the output file report could also be together.
It's probably true for other functions, but they are more difficult to detect.